3303Elevator/src/app/ElevatorSubsystem/Elevator/Elevator.java

```diff
@@ -1,137 +1,238 @@
 		}else {
 			return;
 		}
-		
+
 		ArrayList<Float> waitTime = tms.getElevatorTransitTime(currentFloor, dest);
 		try {
 			Thread.sleep(waitTime.get(0).intValue());
 		}catch(InterruptedException e) {
 			System.out.println("[ERROR]#Elevator#waitTransite() issue waiting");
 		}
 	}
-	
+
 	/**
 	 * Get floor
 	 * */
 	public int getFloor() {
 		return this.currentFloor;
 	}
-	
+
 	/**
 	 * Get state machine for elevator
 	 * */
 	public ElevatorStateMachine getState() {
 		return state;
 	}
-	
+
 	/**
 	 * Update state to next state
 	 * */
 	public void nextState() {
 		state = state.nextState();
-		
+
 		if(state == ElevatorStateMachine.MoveUp) {
 			currentFloor++;
 		}else if(state == ElevatorStateMachine.MoveDown) {
 			currentFloor--;
 		}
 	}
-	
+
 	/**
 	 * Determines if the elevator is in a state that is moving
 	 * @return if the elevator is moving
 	 */
 	public boolean isMoving() {
 		return state == ElevatorStateMachine.MoveUp || state == ElevatorStateMachine.MoveDown;
 	}
-	
+
 	/**
 	 * Determines if the elevator is idle or waiting for next stopping processing
-	 * 
+	 *
 	 * @return if the elevator is currently stationary
 	 */
 	public boolean isStationary() {
 		return state == ElevatorStateMachine.Idle || state == ElevatorStateMachine.NextStopProcessing;
 	}
-	
+
 	/**
 	 * Get the elevator's direction state
 	 * @return the elevator's direction state
 	 */
 	public Direction getDirection() {
 		return state.getDirection();
 	}
-	
+
 	/***
 	 * Set the direction of the state, can not set direction up if on max floor, can not set direction down if on bottom floor.
 	 * @param d Direction
 	 */
 	public void setDirection(Direction d) {
-		
+
 		if(this.currentFloor == 1 && d == Direction.DOWN) {
 			d = Direction.AWAITING_NEXT_REQUEST;
 		}
-		
+
 		if( this.currentFloor == this.maxFloorCount && d == Direction.UP) {
 			d = Direction.AWAITING_NEXT_REQUEST;
 		}
-		
+
 		state.setDirection(d);
 	}
 	
+	/**
+	 * Logs message to logger
+	 * @param message
+	 */
+	public void log(String msg){
+		logger.logElevatorEvents("[Elevator "+ this.id + "]"+msg);
+	}
+	
+	/**
+	 * Checks the destination floor and updates the elevators state
+	 * @param destinationFloor
+	 */
+	private void checkFloor(int destinationFloor) {
+		this.log("at floor " + this.getFloor());
+		if(this.getFloor() == destinationFloor){ //or max floor
+			this.nextState();// stopping or opening door
+			
+			if(this.getState() == ElevatorStateMachine.Stopping) {
+				//elevator slowing down
+				this.nextState();//open door if stopping before
+			}
+			this.log("has arrived at desitnation ");
+			
+			this.log("doors starting to open");
+			this.nextState();//open door
+			this.loadElevator();
+			this.nextState();//door closing
+			this.nextState();
+			this.log("doors finished closing");
+		}
+	}
+	
+	public void move() {
+		// boolean movingUp = elevator.getState() == ElevatorStateMachine.MoveUp;
```

Comment:
No references to move() as of now. Some clarification on comms between elevators, elevatorSubsys, and scheduler will be needed